dashboard/app/controllers/curriculum_tracking_pixel_controller.rb

```diff
@@ -0,0 +1,22 @@
+require 'cdo/firehose'
+
+class CurriculumTrackingPixelController < ApplicationController
+  def index
+    prevent_caching
+    curriculum_page = URI.unescape(params[:from])
+    user_id = current_user&.id
+
+    FirehoseClient.instance.put_record(
```

Comment:
Do you want to protect against writing to firehose if we don't get a valid `:from` parameter?